subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/MavenResolver.java

```diff
@@ -285,7 +285,7 @@ private static boolean isUniqueSnapshot(ModuleComponentIdentifier id) {
     }
 
     protected static boolean isNonUniqueSnapshot(ModuleComponentIdentifier moduleComponentIdentifier) {
-        return moduleComponentIdentifier.getVersion().endsWith("-SNAPSHOT");
+        return moduleComponentIdentifier.getVersion().endsWith("SNAPSHOT");
```

Comment:
```suggestion
        return moduleComponentIdentifier.getVersion().endsWith("-SNAPSHOT") || "SNAPSHOT".equals(moduleComponentIdentifier.getVersion());
```
or
```suggestion
        return moduleComponentIdentifier.getVersion().matches("^(.*-)?SNAPSHOT$");
```

Reply:
Done! Checking SNAPSHOT equality first for better performance.